valetudo-map-card.js

```diff
@@ -835,3 +877,236 @@ class ValetudoMapCard extends HTMLElement {
 }
 
 customElements.define('valetudo-map-card', ValetudoMapCard);
+
+/**
+ * Credit for the following code goes to the authors of Valetudo: https://github.com/Hypfer/Valetudo
```

Comment:
Could you please add the following:

```
This class (FourColorTheoremSolver) is taken from https://github.com/Hypfer/Valetudo/blob/890120c76930bb8941459a7e0d1baa0af8577d83/client/zone/js-modules/map-color-finder.js under the Apache 2 license.
See https://github.com/Hypfer/Valetudo/blob/890120c76930bb8941459a7e0d1baa0af8577d83/LICENSE for more information.
```

Reply:
That way we make it clear exactly what version of the Valetudo code we're using.

Reply:
I added that text to the other classes I pulled from Valetudo as well (MapAreaVertex, MapAreaGraph).